app-auth/graphapponlytutorial/GraphHelper.php

```diff
@@ -3,91 +3,69 @@
 // Licensed under the MIT license.
 
 // <UseSnippet>
-use Microsoft\Graph\Graph;
-use Microsoft\Graph\Http;
-use Microsoft\Graph\Model;
-use GuzzleHttp\Client;
+use Microsoft\Graph\Generated\Models;
+use Microsoft\Graph\Generated\Users\UsersRequestBuilderGetQueryParameters;
+use Microsoft\Graph\Generated\Users\UsersRequestBuilderGetRequestConfiguration;
+use Microsoft\Graph\GraphRequestAdapter;
+use Microsoft\Graph\GraphServiceClient;
+use Microsoft\Kiota\Authentication\Oauth\ClientCredentialContext;
+use Microsoft\Kiota\Authentication\PhpLeagueAuthenticationProvider;
 // </UseSnippet>
 
 class GraphHelper {
     // <AppOnlyAuthConfigSnippet>
-    private static Client $tokenClient;
-    private static string $appToken;
     private static string $clientId = '';
     private static string $clientSecret = '';
     private static string $tenantId = '';
-    private static Graph $appClient;
+    private static PhpLeagueAuthenticationProvider $authProvider;
+    private static GraphServiceClient $appClient;
 
     public static function initializeGraphForAppOnlyAuth(): void {
-        GraphHelper::$tokenClient = new Client();
         GraphHelper::$clientId = $_ENV['CLIENT_ID'];
         GraphHelper::$clientSecret = $_ENV['CLIENT_SECRET'];
         GraphHelper::$tenantId = $_ENV['TENANT_ID'];
-        GraphHelper::$appClient = new Graph();
+
+        $tokenContext = new ClientCredentialContext(
+            GraphHelper::$tenantId,
+            GraphHelper::$clientId,
+            GraphHelper::$clientSecret);
+
+        GraphHelper::$authProvider = new PhpLeagueAuthenticationProvider(
+            $tokenContext,
+            ['https://graph.microsoft.com/.default']);
+
+        $adapter = new GraphRequestAdapter(GraphHelper::$authProvider);
+
+        GraphHelper::$appClient = new GraphServiceClient($adapter);
```

Comment:
Subject to https://github.com/microsoftgraph/msgraph-sdk-php/pull/1241 being merged
```suggestion
        GraphHelper::$appClient = GraphServiceClient::createWithRequestAdapter($adapter);
```